### DISCUSSION TITLE
Using NgIconsModule in SharedModules

### DISCUSSION THREAD

**Discussion by RobertoHixsa** [ng-icons/ng-icons#79] [Q&A] [answered]
Hi, I can import Icons using the NgIconsModule without any problem but I can't export them. How can I use them in this way?
![sharedmodule](https://github.com/ng-icons/ng-icons/assets/143147489/6cd7bc39-706e-46f5-a2ea-d2c620a454c1)
![sharedmodule2](https://github.com/ng-icons/ng-icons/assets/143147489/c9923df9-2a1f-4483-b4f2-48414291776b)

**Comment by ashley-hunter** [comment, marked as answer]
Hi, sure, so as you discovered you cannot export them from an NgModule, the reason for that is that NgModules can only export components. Our icons are not components but instead are strings that are registered in Angular's dependency injection system. With the way the DI system works this makes icons available to any child components (or lazy loaded modules) without any need to export.

To make icons available to your entire application you can either register them in the AppModule:

```ts
@NgModule({
    imports: [NgIconsModule.withIcons(MaterialIcons)]
})
export AppModule {}
```

Or have a `CoreModule` that your AppModule imports. This module typically contains anything that registers things with the DI system  e.g.:

```ts
@NgModule({
    imports: [NgIconsModule.withIcons(MaterialIcons)]
})
export CoreModule {}


@NgModule({
    imports: [CoreModule]
})
export AppModule {}
```

If you want however you can still export the NgIcon component in your shared module, this will allow any components with access to shared module access to use `<ng-icon>` in their templates:

```ts
@NgModule({
    imports: [NgIconsModule],
    exports: [NgIconsModule],
})
export SharedModule {}
```

If anything is unclear let me know and I will try and clarify further.

**Reply by RobertoHixsa** [reply]
Thanks a lot for your help. Now everything is clear!

**Reply by ashley-hunter** [reply]
Excellent! No problem!
